ddtrace/contrib/pytest_bdd/plugin.py

```diff
@@ -0,0 +1,90 @@
+import json
+import os
+
+import pytest
+
+from ddtrace.contrib.pytest.plugin import _extract_span as _extract_feature_span
+from ddtrace.contrib.pytest_bdd.constants import STEP_KIND
+from ddtrace.ext import test
+from ddtrace.pin import Pin
+
+
+def _extract_span(item):
+    """Extract span from `step_func`."""
+    return getattr(item, "_datadog_span", None)
+
+
+def _store_span(item, span):
+    """Store span at `step_func`."""
+    setattr(item, "_datadog_span", span)
+
+
+def pytest_sessionstart(session):
+    if session.config.pluginmanager.hasplugin("pytest-bdd"):
+        session.config.pluginmanager.register(_PytestBddPlugin, "_datadog-pytest-bdd")
+
+
+class _PytestBddPlugin:
+    @staticmethod
+    @pytest.hookimpl(tryfirst=True)
+    def pytest_bdd_before_scenario(request, feature, scenario):
+        pin = Pin.get_from(request.config)
+        if pin:
+            span = _extract_feature_span(request.node)
+            if span is not None:
+                location = os.path.relpath(scenario.feature.filename, str(request.config.rootdir))
+                span.set_tag(test.NAME, scenario.name)
+                span.set_tag(test.SUITE, location)
```

Comment:
this is extracting the span from the `pytest` integration, right? Are we overwriting the name and suite coming from the `pytest` instrumentation? Or will they be empty?

Reply:
Yes, we are overriding the name. In this case it desirable as the name coming from `pytest` integration points to an internal file in `pytest_bdd` package since the tests are generated on-the-fly.